addons/mail/static/src/discuss/core/web/discuss_sidebar_categories.js

```diff
@@ -1,63 +1,123 @@
 /* @odoo-module */
 
 import { ImStatus } from "@mail/core/common/im_status";
-import { useMessaging, useStore } from "@mail/core/common/messaging_hook";
+import { useStore } from "@mail/core/common/messaging_hook";
 import { ThreadIcon } from "@mail/core/common/thread_icon";
+import { discussSidebarItemsRegistry } from "@mail/core/web/discuss_sidebar";
+import { ChannelSelector } from "@mail/discuss/core/web/channel_selector";
 import { onExternalClick } from "@mail/utils/common/hooks";
-import { markEventHandled } from "@web/core/utils/misc";
 
 import { Component, useState } from "@odoo/owl";
 
 import { ConfirmationDialog } from "@web/core/confirmation_dialog/confirmation_dialog";
 import { _t } from "@web/core/l10n/translation";
+import { registry } from "@web/core/registry";
 import { useService } from "@web/core/utils/hooks";
+import { markEventHandled } from "@web/core/utils/misc";
+
+export const discussSidebarCategoriesRegistry = registry.category(
+    "mail.discuss_sidebar_categories"
+);
+export const discussSidebarChannelIndicatorsRegistry = registry.category(
+    "mail.discuss_sidebar_channel_indicators"
+);
```

Comment:
Could be nice to have a registry for all thread commands in sidebar, including settings and leave/unpin... Or maybe use `thread_actions` and allow to customise context to show them in sidebar?

Not necessarily to do in this PR. Just a random idea!